frodo.js

```diff
@@ -1,28 +1,27 @@
 /**
```

Comment:
SCRIPT-I'm impressed with your use of design patterns here, it really enhances the code organization.